toolpathing/FillPathScheduler2d.cs

```diff
@@ -62,40 +62,61 @@ public SequentialScheduler2d(ToolpathSetBuilder builder, SingleMaterialFFFSettin
 
 
 		// [TODO] no reason we couldn't start on edge midpoint??
-		public virtual void AppendPolygon2d(FillPolygon2d poly) {
-			Vector3d currentPos = Builder.Position;
-			Vector2d currentPos2 = currentPos.xy;
+		public virtual void AppendPolygon2d(FillPolygon2d poly)
+        {
+            Vector3d currentPos = Builder.Position;
+            Vector2d currentPos2 = currentPos.xy;
 
-			int N = poly.VertexCount;
-			if (N < 2)
-				throw new Exception("PathScheduler.AppendPolygon2d: degenerate curve!");
+            int N = poly.VertexCount;
+            if (N < 2)
+                throw new Exception("PathScheduler.AppendPolygon2d: degenerate curve!");
 
-			int iNearest = CurveUtils2.FindNearestVertex(currentPos2, poly.Vertices);
+            int iNearest = CurveUtils2.FindNearestVertex(currentPos2, poly.Vertices);
 
-			Vector2d startPt = poly[iNearest];
+            Vector2d startPt = poly[iNearest];
 
-            // a travel may require a retract, which we might want to skip
-            if (ExtrudeOnShortTravels && currentPos2.Distance(startPt) < ShortTravelDistance )
-                Builder.AppendExtrude(startPt, Settings.RapidTravelSpeed);
-            else
-			    Builder.AppendTravel(startPt, Settings.RapidTravelSpeed);
+            AppendTravel(currentPos2, startPt);
 
-			List<Vector2d> loopV = new List<Vector2d>(N + 1);
-			for (int i = 0; i <= N; i++ ) {
-				int k = (iNearest + i) % N;
-				loopV.Add(poly[k]);
-			}
+            List<Vector2d> loopV = new List<Vector2d>(N + 1);
+            for (int i = 0; i <= N; i++)
+            {
+                int k = (iNearest + i) % N;
+                loopV.Add(poly[k]);
+            }
 
             double useSpeed = select_speed(poly);
 
-			Builder.AppendExtrude(loopV, useSpeed, poly.TypeFlags, null);
-		}
+            Builder.AppendExtrude(loopV, useSpeed, poly.TypeFlags, null);
+        }
+
+        private void AppendTravel(Vector2d startPt, Vector2d endPt)
```

Comment:
Broke out into new method:
 - original logic for whether or not to retract
 - new logic for whether or not to do a z-lift before moving